semgrep/semgrep/semgrep_main.py

```diff
@@ -109,6 +110,9 @@ def rule_match_nosem(rule_match: RuleMatch, strict: bool) -> bool:
         for pattern_id in COMMA_SEPARATED_LIST_RE.split(ids_str)
         if pattern_id.strip()
     }
+    # Filter out ids that are not alphanum+dashes+underscores.
+    # This removes trailing symbols from comments, such as HTML comments' `-->`.
+    pattern_ids = set(filter(lambda x: not sub(r"[\w-]+", "", x), pattern_ids))
```

Comment:
It'd probably be better to modify `NOSEM_INLINE_RE`, if possible, to allow arbitrary trailing symbols after the pattern ID list. This would also be helpful for C-like languages using something like:

```c
statement; /* nosem: pattern-id1, pattern-id2 */
```

Or even Python's

```python
statement  '''nosem: pattern-id1, pattern-id2'''
```

Although I'm not sure how you distinguish `*/` from being included as a pattern ID, or `pattern-id2'''` being used instead of `pattern-id2`. :thinking:

Another thing to consider: I think this might be the first time we're taking a hard stance that ids must be alphanum+dashes+underscores. Until now I think it's been any string. We certainly can make this requirement, but there could be downstream effects (user's existing `nosem` functionality breaking), and this may be inconsistent with other parts of the Semgrep codebase.

Reply:
I was intimidated by that RE and didn't wanna touch it 😅 

This is the "right" approach, though, so I'll take another crack at it tomorrow.

Reply:
I think it's going to be difficult to handle this case without explicitly looking for comment-ending syntax. If I'm thinking about this correctly, comment syntaxes are, by definition, unique character combinations, which means that are difficult to find with regexes in a generic fashion.

Some options here include:
1) specifying a rigid format for nosem comments, such as not allowing any spaces or using the current implementation of removing things that don't look like rule IDs.
2) explicitly removing known ending comment syntaxes.

I'm going forward with (2) because I'm afraid of unintended consequences like you pointed out, @mschwager . I also worry about "guessing" at rule IDs because I would implement a simple "does this have a readable character" filter, but can imagine edge cases where someone may not want readable characters as rule IDs. And what about Unicode? 😅 

(2) is crude but because comment endings are so unique, I think it will avoid problems for a while. It just has the downside of having to be kept up-to-date.

I am curious about @aryx's or @mjambon's or @emjin's thoughts on the matter, since they are the PA experts and all. :)

Reply:
I would vote for (1), simple regexp like \bnosem:\s*[\w-_]+

Reply:
I decided to stick with the original filtering implementation, since it's a bit cleaner than (2) and (1) still nabs things like `--` from `-->`.

lmk if you think we should revisit this.